### PR TITLE
Issue #74: No Such Element Exception raised during windows collect

### PR DESCRIPTION
**Pull request by CherfaElyes**
* Fixed the NoSuchElementException
* Removed duplicated connector detection criteria validation

### CODE REVIEW

**Comment by CherfaElyes** [comment]
image

File: metricshub-agent/src/main/java/org/sentrysoftware/metricshub/cli/service/PrettyPrinterService.java [hunk 1]
```diff
@@ -186,65 +186,67 @@ private void print(final MonitorChildren monitorChildren, final int indentation)
...
+			printWriter.println();
+
+			// Add the margin
+			addMargin(indentation);
+
+			// Add the header
+			printWriter.println("Metrics:");
+
+			// Get the metrics
+			final Map<String, AbstractMetric> metrics = monitor.getMetrics();
```
Comment (NassimBtk) [inline metricshub-agent/src/main/java/org/sentrysoftware/metricshub/cli/service/PrettyPrinterService.java:200]: `monitor.getMetrics()` is called several times, create a variable please.

File: metricshub-agent/src/main/java/org/sentrysoftware/metricshub/cli/service/PrettyPrinterService.java [hunk 2]
```diff
@@ -186,65 +186,67 @@ private void print(final MonitorChildren monitorChildren, final int indentation)
 	 * @param indentation How much indentation to use
 	 */
 	void printMetrics(final Monitor monitor, final int indentation) {
-		// Print a new line
-		printWriter.println();
-
-		// Add the margin
-		addMargin(indentation);
+		if (monitor.getMetrics() != null && !monitor.getMetrics().isEmpty()) {
```
Comment (NassimBtk) [inline metricshub-agent/src/main/java/org/sentrysoftware/metricshub/cli/service/PrettyPrinterService.java:189]: To verify metrics, I would use the following approach:
```java
		// Get the metrics
		final Map<String, AbstractMetric> metrics = monitor.getMetrics();

		if (metrics == null || metrics.isEmpty()) {
			return;
		}
```

**Review by NassimBtk — CHANGES_REQUESTED** [review]
Minor changes and we are good to go.

**Review by NassimBtk — APPROVED** [review]

**Pull request merged**